pkg/controller/controller_instance.go

```diff
@@ -1207,6 +1207,13 @@ func (c *controller) updateServiceInstanceReferences(toUpdate *v1beta1.ServiceIn
 // Note: objects coming from informers should never be mutated; the instance
 // passed to this method should always be a deep copy.
 func (c *controller) updateServiceInstanceStatus(instance *v1beta1.ServiceInstance) (*v1beta1.ServiceInstance, error) {
+	return c.updateServiceInstanceStatusWithRetries(instance, nil)
+}
+
+func (c *controller) updateServiceInstanceStatusWithRetries(
+	instance *v1beta1.ServiceInstance,
+	updateFunc func(*v1beta1.ServiceInstance)) (*v1beta1.ServiceInstance, error) {
```

Comment:
I'd certainly benefit from a comment on this function indicating the updateFunc (if specified) applies additional updates to the instance which must be applied in the event this instance is out of date and an updated instance must be used. 

LGTM though.

Reply:
Agreed, this really needs more explanation and perhaps a different name? If I understand this properly the function is applied only when retrying after an conflict, a more clear name would perhaps be `resolveConflictFunc` or something along those lines.

Reply:
I've added a comment & renamed the function to `postConflictUpdateFunc`.